app/src/main/java/com/kamwithk/ankiconnectandroid/routing/AnkiAPIRouting.java

```diff
@@ -159,19 +177,41 @@ private String canAddNotes(JsonObject raw_json) {
         return Parser.gson.toJson(integratedAPI.noteAPI.canAddNotes(notes_to_test));
     }
 
+    /**
+     * Add a new note to Anki.
+     * The note can include audio files, which will be downloaded. Only downloadable audio files are currently supported.
+     */
     private String addNote(JsonObject raw_json) throws Exception {
-        return String.valueOf(integratedAPI.addNote(
-                Parser.getNoteValues(raw_json),
+        Map<String, String> noteValues = Parser.getNoteValues(raw_json);
+
+        List<DownloadAudioRequest> audioRequests = Parser.getDownloadAudioRequests(raw_json);
+        for (DownloadAudioRequest audioRequest : audioRequests) {
+            // download the requested audio file to the media folder
+            String filePath = integratedAPI.downloadAndStoreAudioFile(audioRequest);
+
+            // attach this audio to the note.
+            // Note that the field contents are replaced with just the audio.
+            String[] fields = audioRequest.getFields();
+            for (String field : fields) {
+                noteValues.put(field, "[sound:" + filePath + "]");
+            }
+        }
+
+        String noteId = String.valueOf(integratedAPI.addNote(
+                noteValues,
                 Parser.getDeckName(raw_json),
                 Parser.getModelName(raw_json),
                 Parser.getNoteTags(raw_json)
         ));
+
+        return noteId;
     }
 
     private String storeMediaFile(JsonObject raw_json) throws Exception {
-        return Parser.gson.toJson(integratedAPI.storeMediaFile(
-                Parser.getMediaFilename(raw_json),
-                Parser.getMediaData(raw_json)
-        ));
+        AudioFile audioFile = new AudioFile();
+        audioFile.setFilename(Parser.getMediaFilename(raw_json));
+        audioFile.setData(Parser.getMediaData(raw_json));
+
+        return Parser.gson.toJson(integratedAPI.storeMediaFile(audioFile));
```

Comment:
Fortunately, these changes still work on images (I tested this on a Yomichan dictionary with images, and it exported perfectly fine). However, changing this is a misnomer, because the media can be images or audio (and theoretically, anything other byte array). I think it would be safe to undo the logic, or rename the `AudioFile` class to something like `BinaryFile`.